src/main/java/org/joshsim/lang/interpret/visitor/JoshConfigParserVisitor.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * ANTLR visitor for parsing Josh configuration files (.jshc format).
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.joshsim.lang.interpret.visitor;
+
+import java.math.BigDecimal;
+import org.joshsim.engine.config.ConfigBuilder;
+import org.joshsim.engine.value.converter.Units;
+import org.joshsim.engine.value.engine.EngineValueFactory;
+import org.joshsim.engine.value.type.EngineValue;
+import org.joshsim.lang.antlr.JoshConfigBaseVisitor;
+import org.joshsim.lang.antlr.JoshConfigParser;
+import org.joshsim.lang.interpret.fragment.jshc.JshcConfigBuilderFragment;
+import org.joshsim.lang.interpret.fragment.jshc.JshcFragment;
+import org.joshsim.lang.interpret.fragment.jshc.JshcValueFragment;
+
+/**
+ * ANTLR visitor for parsing .jshc configuration files.
+ *
+ * <p>This visitor processes ANTLR parse trees generated from .jshc files
+ * and produces JshcFragment instances containing the parsed configuration data.</p>
+ */
+public class JoshConfigParserVisitor extends JoshConfigBaseVisitor<JshcFragment> {
```

Comment:
Across this, let's use `ctx.*.accept` instead of calling the visit directly similar to `JoshMathematicsVisitor` and others.